### ISSUE TITLE
Breaking change not detected

### ISSUE THREAD

**Issue by relekang** [relekang/python-semantic-release#200]
The v5 release had to be done manually with `--major` because the automatic release kept tagging it as v4.12.0.

**Comment by AlphaMycelium** [comment]
Note: the relevant commit for this is https://github.com/relekang/python-semantic-release/commit/740f4bdb26569362acfc80f7e862fc2c750a46dd

**Comment by AlphaMycelium** [comment]
Here is the text of commit 740f4bd:

```
feat(build): allow config setting for build command (#195)

* feat(build): allow config setting for build command

BREAKING CHANGE: Previously the build_commands configuration variable
set the types of bundles sent to `python setup.py`. It has been replaced
by the configuration variable `build_command` which takes the full
command e.g. `python setup.py sdist` or `poetry build`.

Closes #188
```

Look at this code from `parse_text_block`:

https://github.com/relekang/python-semantic-release/blob/3baf55bf425b340e1be12a70868526066c9708b6/semantic_release/history/parser_helpers.py#L18-L24

740f4bd has the commit message followed by 3 paragraphs of extended description. This means that the if statement on L21 evaluates to false (as 3 is not equal to 2), so the 'footer' is not considered, despite it still existing.

https://github.com/relekang/python-semantic-release/blob/3baf55bf425b340e1be12a70868526066c9708b6/semantic_release/history/parser_helpers.py#L10-L12

The above docstring is wrong, 'footer' is actually defined as the *second* paragraph.

Also, with squashed commits, there are often many paragraphs in the description, which means that the breaking change (or multiple breaking changes) could appear later on.

I recommend that we change the way these breaking changes are searched for so that instead we loop over all paragraphs. This could possibly be handled entirely in the commit parser itself, adding a list of breaking changes as the fifth item in the returned tuple. That would also allow for future alternative parsers to find breaking changes differently.

**Comment by codejedi365** [comment]
Is this still working?  I found a squashed merge that has a breaking change in it was not detected.  It was detected previously when it was on a separate branch.  After the squash, it failed to be detected.
